lightning/src/offers/async_receive_offer_cache.rs

```diff
@@ -374,9 +397,8 @@ impl AsyncReceiveOfferCache {
 
 		// Filter for unused offers where longer than OFFER_REFRESH_THRESHOLD time has passed since they
 		// were last updated, so they are stale enough to warrant replacement.
-		let awhile_ago = duration_since_epoch.saturating_sub(OFFER_REFRESH_THRESHOLD);
 		self.unused_ready_offers()
-			.filter(|(_, offer, _)| offer.created_at < awhile_ago)
+			.filter(|(_, offer, _)| offer.needs_replacement(duration_since_epoch))
```

Comment:
This is now double-filtering for ready? Maybe self.offers_with_idx() can be used.